streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -735,11 +728,15 @@ public Serde keySerde() {
      */
     public Serde defaultKeySerde() {
         try {
-            Serde<?> serde = getConfiguredInstance(DEFAULT_KEY_SERDE_CLASS_CONFIG, Serde.class);
+            Serde<?> serde = getConfiguredInstance(KEY_SERDE_CLASS_CONFIG, Serde.class);
+            if (serde == null) {
+                serde = getConfiguredInstance(DEFAULT_KEY_SERDE_CLASS_CONFIG, Serde.class);
+            }
             serde.configure(originals(), true);
```

Comment:
Why do we need to call `configure` -- should this not have happened with `getConfiguredInstance` already?